### PR TITLE
refactor: unify http client creation

### PR DESCRIPTION
**Pull request by h0lyalg0rithm**
## Proposed changes

Unifies the httpclient used across the application.
No validation of server certificate

## Types of changes

What types of changes does your code introduce to the repo? Put an `x` in the boxes that apply
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices applies)

## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.
- [x] I have signed the commit for DCO to be passed.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Review by glimchb — CHANGES_REQUESTED** [review]
maybe i miss something....
why not just `daemon.downloadAndValidateImage()` call to `utils.doTLSRequest()` ?

the method post vs get can be an argument 

happy to discuss ....

**Comment by h0lyalg0rithm** [comment]
> maybe i miss something.... why not just `daemon.downloadAndValidateImage()` call to `utils.doTLSRequest()` ?
> 
> the method post vs get can be an argument
> 
> happy to discuss ....

@glimchb I kept them seperate because there were some extra http headers (basic auth etc) being passed in the post request, no sure if that was needed in the get version.
I had to change inject the http client to enable testing for the caching feature.

File: sztp-agent/pkg/secureagent/agent.go [hunk 1]
```diff
@@ -171,3 +184,25 @@ func (a *Agent) SetContentTypeReq(ct string) {
 func (a *Agent) SetProgressJSON(p ProgressJSON) {
 	a.ProgressJSON = p
 }
+
+func NewHttpClient(bootstrapTrustAnchorCert string, deviceEndEntityCert string, devicePrivateKey string) http.Client {
+	caCert, _ := os.ReadFile(bootstrapTrustAnchorCert)
```
Comment (glimchb) [inline sztp-agent/pkg/secureagent/agent.go:189]: let's move this from `agent` to `tls` ?

File: sztp-agent/pkg/secureagent/agent.go [hunk 2]
```diff
@@ -68,6 +75,11 @@ type BootstrapServerErrorOutput struct {
 	} `json:"ietf-restconf:errors"`
 }
 
+type HttpClient interface {
```
Comment (glimchb) [inline sztp-agent/pkg/secureagent/agent.go:78]: let's move this from `agent` to `tls` ?

**Comment by glimchb** [comment]
> I had to change inject the http client to enable testing for the caching feature.

so this should be separate PR or separate commit just for injection

File: sztp-agent/pkg/secureagent/agent.go (hunk 2 above)
Comment (glimchb) [reply sztp-agent/pkg/secureagent/agent.go:78]: ok, so you want interface here - fine
but then the object itself - http or ftp in their own files... ok ?

Comment (h0lyalg0rithm) [reply sztp-agent/pkg/secureagent/agent.go:78]: Yes exactly.Have separate files for each client

**Review by glimchb — COMMENTED** [review]
so `NewHttpClient()` should move to another file - right ?

**Review by glimchb — APPROVED** [review]
lgtm

**Comment by bhoopesh369** [comment]
LGTM
if @h0lyalg0rithm is away for now, I'd be happy to cherry-pick the changes and open a new PR.

**Pull request merged**